### PR TITLE
chore: upgrade to 1.21

### PR DESCRIPTION
**Pull request by codyoss**

### CODE REVIEW

File: .github/workflows/vet.yml [hunk 1]
```diff
@@ -9,14 +9,17 @@ on:
...
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
     - uses: actions/setup-go@v5
       with:
-        go-version: '1.20.x'
+        go-version: '1.21.x'
     - name: Install tools
       run: |
         go install golang.org/x/lint/golint@latest && \
```
Comment (quartzmo) [inline .github/workflows/vet.yml:25]: Can you restore `go install honnef.co/go/tools/cmd/staticcheck@v0.4.7` to `go install honnef.co/go/tools/cmd/staticcheck@latest` on line 27, below? Thanks!

File: internal/generated/snippets/go.mod [hunk 2]
```diff
@@ -1,6 +1,8 @@
 module cloud.google.com/go/internal/generated
 
-go 1.20
+go 1.21
+
+toolchain go1.21.13
```
Comment (quartzmo) [inline internal/generated/snippets/go.mod:5]: Just curious, why does this go.mod get `toolchain`, when others do not?

File: spanner/test/opentelemetry/test/go.mod [hunk 3]
```diff
@@ -1,6 +1,8 @@
 module cloud.google.com/go/spanner/test
 
-go 1.20
+go 1.21
+
+toolchain go1.21.13
```
Comment (quartzmo) [inline spanner/test/opentelemetry/test/go.mod:5]: Same question here also, why `toolchain`?

File: storage/internal/benchmarks/go.mod [hunk 4]
```diff
@@ -1,6 +1,8 @@
 module main
 
-go 1.20
+go 1.21
+
+toolchain go1.21.13
```
Comment (quartzmo) [inline storage/internal/benchmarks/go.mod:5]: Same question here also, why `toolchain`?

File: spanner/test/opentelemetry/test/go.mod (hunk 3 above)
Comment (codyoss) [reply spanner/test/opentelemetry/test/go.mod:5]: same as above.

File: storage/internal/benchmarks/go.mod (hunk 4 above)
Comment (codyoss) [reply storage/internal/benchmarks/go.mod:5]: same as above.

File: .github/workflows/vet.yml (hunk 1 above)
Comment (codyoss) [reply .github/workflows/vet.yml:25]: The latest staticcheck requires 1.22: https://github.com/dominikh/go-tools/blob/56172d41b117cc2c2f99f65fe0a790c8d7d7ea66/go.mod#L3

One thing we could consider is only running vet on latest version. We do this in other repos today.

File: internal/generated/snippets/go.mod (hunk 2 above)
Comment (codyoss) [reply internal/generated/snippets/go.mod]: good catch, I think I used my wrong tidy alias here. tidied again and gone.

**Review by shollyman — APPROVED** [review]

File: .github/workflows/vet.yml (hunk 1 above)
Comment (quartzmo) [reply .github/workflows/vet.yml:25]: Until now, I thought it was good to run vet on the oldest version in order to keep `toolchain` from sneaking into go.mod files in our repo. But maybe we no longer have any oldest version concerns like that.

Comment (quartzmo) [reply .github/workflows/vet.yml:25]: > The latest staticcheck requires 1.22

See you in 2025, staticcheck!

**Review by quartzmo — APPROVED** [review]

**Pull request merged**